### ISSUE TITLE
[P2] RNNのBPTT（時間方向の誤差逆伝播）を視覚化するデモを実装する

### ISSUE THREAD

**Issue by tomiya7688** [tomiya7688/comfyUI_support_tools#118]
## 目的
RNNの学習で使われる **Backpropagation Through Time (BPTT)** を、時間展開したRNN上で誤差・勾配が過去stepへ伝播する様子として視覚化する。

#105 の通常の誤差逆伝播から、系列モデル特有の「時間方向へ同じ重みを通じて勾配が流れる」学習へ橋渡しする。

## 依存
- #105 誤差逆伝播法デモ
- #113 RNN基本デモ

## 実装内容
- [ ] RNNを時間方向へunroll表示
- [ ] 各時刻のlossを表示
- [ ] backward stepを時間逆順に表示
- [ ] hidden-to-hidden weightへ複数時刻からgradientが加算される様子を表示
- [ ] 各stepのgradient magnitudeを表示
- [ ] sequence lengthを変更可能にする
- [ ] truncated BPTTの簡易切替を用意
- [ ] gradient clipping ON/OFFを比較可能にする
- [ ] 1 Backward Step / Run / Reset

## 視覚フロー
```text
Forward:
x1 -> h1 -> h2 -> h3 -> ... -> loss

Backward:
loss -> h3 -> h2 -> h1
          ↓     ↓     ↓
       shared recurrent weights
```

## 教育用説明
- 「RNNでは同じ重みを時間方向に繰り返し使うため、勾配も複数stepを通って伝わります」
- 「系列が長くなると勾配が小さくなりすぎたり大きくなりすぎたりすることがあります」
- 「LSTM/GRUは長期依存を扱いやすくするための構造として導入されます」

## テスト
- [ ] 時間展開した手計算gradientと一致する
- [ ] shared weight gradientが各時刻の寄与の和になる
- [ ] truncated BPTTで指定範囲外へgradientが流れない
- [ ] clipping有効時にgradient normが上限以内になる

## 完了条件
- 時間方向のforward/backwardを別々に追える
- shared recurrent weightへのgradient蓄積を確認できる
- vanishing / exploding gradientの概念を数値と視覚で確認できる
- #114 LSTM/GRUへ自然につながる

## 関連
- #97 活性化関数比較
- #114 LSTM / GRU比較

## Out of Scope
- RNN自動微分エンジン
- 大規模sequence training
- Transformer backprop
- GPU最適化
